### PR TITLE
Introduce sticky footer

### PR DESCRIPTION
**Pull request by dgalarza**
* Adds support for `gap` prop to `Box` component

image

### CODE REVIEW

File: packages/components/sticky_footer/src/sticky_footer.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  right: 0,
+  bottom: 0,
+}))<FlexProps>``;
+
+StickyFooter.defaultProps = {
+  gap: "m",
+  zIndex: 2,
+  alignItems: "center",
+  justifyContent: "flex-end",
+};
```
Comment (acostanzo) [inline packages/components/sticky_footer/src/sticky_footer.tsx:20]: I'm not sure I follow what's happening here. Why are some items in default Props and others in the attrs call? Ultimately, should we consider moving to more of a composition of Flex vs leveraging `styled`? e.g.
```suggestion
interface OwnProps {
  children: React.ReactNode;
}

export type StickyFooterProps = OwnProps;

export const StickyFooter: React.FC<StickyFooterProps> = ({
  children,
}): React.ReactElement => {
  return (
    <Flex
      alignItems="center"
      bg="white"
      borderTop="1SolidSubdued"
      bottom={0}
      gap="m"
      height="stickyFooter"
      justifyContent="flex-end"
      left={0}
      position="fixed"
      px="xxxl"
      right={0}
      zIndex={2}
    >
      {children}
    </Flex>
  );
};
```

If need be you could include and spread remaining Flex Props.

Comment (dgalarza) [reply packages/components/sticky_footer/src/sticky_footer.tsx:20]: There's a couple things at play here. To answer your last question with regards to composing from Flex, that's essentially what we are doing by using `styled(Flex)`. https://styled-components.com/docs/basics#extending-styles

As for `attrs` vs `defaultProps`, the `attrs` method allows us to attach "static" props to the component which means they cannot be changed, as opposed to the `defaultProps` which provides some defaults which can be altered by the consumer.

https://styled-components.com/docs/basics#attaching-additional-props

Comment (acostanzo) [reply packages/components/sticky_footer/src/sticky_footer.tsx:20]: Ah I see, so in this case you wouldn't be able to pass height? Does typescript know this or does it still allow you to put height on the component, and just not obey it?

Comment (dgalarza) [reply packages/components/sticky_footer/src/sticky_footer.tsx]: Good call out @acostanzo.

So using `attrs` did not remove the prop from typescript autocompletion which leads to confusing results. I've opted to move to away from using `styled` in favor of a regular component. This did lead me down the rabbit hole of figuring out the complaint of trying to splat props down to a styled-component with styled-system. I was able to hunt down the issue via this thread / solution - https://github.com/styled-system/styled-system/issues/1206#issuecomment-1014609863

This should remove the biggest issue we've seen to date when trying to simply accept and proxy styled-component props down to another component. This is ready for another review.

**Review by acostanzo — APPROVED** [review]
Well I'll be damned I guess that's why after all this time... sometimes it worked and sometimes it didn't! Never would have guessed `as` had anything to do with it. I wonder if under the hood maybe there is some typing related to standard element type props and if it's not explicit maybe it goes to unknown 🤔

**Pull request merged**